src/pyfx/view/json_lib/selector/__init__.py

```diff
@@ -0,0 +1,41 @@
+from numbers import Number
+
+from ..object import ObjectNode
+from ..array import ArrayNode
+from ..primitive import StringNode, IntegerNode, NumericNode, BooleanNode, NullNode, GenericNode
+
+def hasattrs(x,*attrs):
+    return all(hasattr(x,attr) for attr in attrs)
+
+def is_numeric(x):
+    return isinstance(x,Number) or isinstance(x,type) and issubclass(x,Number)
+
+def is_dictlike(x):
+    return hasattrs(x,'__getitem__','__iter__','__len__','keys')
+
+def is_arraylike(x):
+    return hasattrs(x,'__getitem__','__iter__','__len__')
+
+
+class DefaultImplementationSelector:
+    def __getitem__(self,key):
```

Comment:
I checked out your branch and ran the tests locally, it seems to me the reason it does not work anymore is this `DefaultImplementationSelector` implementation is not considered as a `dict` in Python but a `list`.

FYI, this is where it fails, https://github.com/cielong/pyfx/blob/365e9906a7db4aa3aaba34c54a98470a48a08e41/src/pyfx/view/json_lib/node_factory.py#L16.

For some reason, while in the `in` operator, the parameter `key` is converted into a numeric, resulting in an infinite loop in Python.